src/components/organisms/LocalSearch.tsx

```diff
@@ -0,0 +1,932 @@
+              onClick={toggleRegexSearch}
+            >
+              <Icon path={mdiRegex} />
+            </LocalSearchStyledButton>
+          </SearchOptionsInnerContainer>
+        </LocalSearchInputLeft>
+        <LocalSearchInputRightContainer>
+          <NumResultsContainer>
+            {numberOfFoundItems > 0 ? (
+              numberOfFoundItems < LOCAL_SEARCH_MAX_RESULTS ? (
+                `${selectedItemIndex + 1}/${numberOfFoundItems}`
+              ) : (
+                '10,000+'
+              )
+            ) : (
+              <ResultStatusContainer
+                className={searchResultError ? 'error' : ''}
+              >
+                {regexSearch && searchResultError ? 'Bad pattern' : '0 results'}
+              </ResultStatusContainer>
+            )}
+          </NumResultsContainer>
+          <SearchOptionsOuterContainer>
+            <SearchNavOptions>
+              <LocalSearchStyledButton
+                disabled={numberOfFoundItems === 0}
+                title={'Previous Occurrence (Shift+F3)'}
+                className={'button'}
+                onClick={() => navigateToNextItem('previous')}
+              >
+                <Icon path={mdiArrowUp} />
+              </LocalSearchStyledButton>
+              <LocalSearchStyledButton
+                disabled={numberOfFoundItems === 0}
+                title={'Next Occurrence (F3)'}
+                className={'button'}
+                onClick={() => navigateToNextItem('next')}
+              >
+                <Icon path={mdiArrowDown} />
+              </LocalSearchStyledButton>
+            </SearchNavOptions>
+          </SearchOptionsOuterContainer>
+          <LocalSearchInputRightClose>
+            <LocalSearchStyledButton
+              className={'button'}
+              onClick={onSearchClose}
+            >
+              <Icon path={mdiClose} />
+            </LocalSearchStyledButton>
+          </LocalSearchInputRightClose>
+        </LocalSearchInputRightContainer>
+      </SearchResultItem>
+      {showingReplace && (
+        <LocalReplace
+          searchOptions={searchOptions}
+          codeMirror={codeMirror}
+          replaceQuery={replaceQuery}
+          numberOfFoundItems={numberOfFoundItems}
+          focusingReplace={focusingReplace}
+          navigateToNext={(direction) => {
+            navigateToNextItem(direction, false)
+            setFocusingReplace(true)
+          }}
+          onReplaceToggle={handleOnReplaceToggle}
+          onReplaceQueryChange={onReplaceQueryChange}
+          onReplacementFinished={handleOnReplacementFinished}
+          onFocusSearchInput={focusSearchTextAreaInput}
+          onReplaceClose={handleOnReplaceClose}
+          onUpdateSearchOptions={onUpdateSearchOptions}
+        />
+      )}
+    </LocalSearchContainer>
+  )
+}
+
+export default LocalSearch
+
+export interface LocalSearchTextAreaProps {
+  numRows: number
+}
+
+export const SearchResultItem = styled.div`
```

Comment:
Extract all exported components to components/atoms or components/molecules.

Reply:
Refactored, 
Not sure if I would do it for SearchResultItem into a component and just pass children (because the only style is reused) or just extract this style to the styled component...

So I refactored the button to component and this only to separate style, let me know if this is correct